content/en/blog/_posts/2021-08-06-api-server-tracing.md

```diff
@@ -0,0 +1,30 @@
+---
+layout: blog
+title: 'API Server Tracing'
+date: 2021-08-06
+slug: api-server-tracing
+---
+
+**Authors:** David Ashpole (Google)
+
+In distributed systems, it can be hard to figure out where problems are. You grep through one components' logs just to discover that the source of your problem is in another component.  You search there only to discover that you need to enable debug logs to figure out what really went wrong... And it goes on. The more complex the path your request takes, the harder it is to answer questions about where it went.  I've personally spent many hours doing this dance with a variety of kubernetes components.
+
+## What is Tracing?
+
+Tracing is supposed to help with these cases.  Whereas logs are grouped together for a given component, traces are grouped together by the request that caused them.  Logging limits the amount of data ingested by using log levels, which means details are often missing.  Tracing uses sampling to collect fewer, but generally more detailed traces. My favorite aspect, though, are how useful the visualisations of traces are.  As we will see, even if you don't understand the inner workings of the API Server, or don't have a clue what an Etcd "Transaction" is, I'd wager you can interpret the trace below (Spoiler!), and would have a pretty good guess for where the problem lies.
```

Comment:
I would change "My favorite aspect, though, are" to "My favorite aspect, though, is". I know that in the UK people often refer to things in plural where we would refer in the singular. Changing "aspect" to "aspects" could work as well.

Reply:
Updated